examples/cpp-simple/server.cc

```diff
@@ -16,101 +16,115 @@
-    std::cout << "C++ Game Server has started!" << std::endl;
-
-    std::cout << "Getting the instance of the SDK!" << std::endl;
-    agones::SDK *sdk = new agones::SDK();
-
-    std::cout << "Attempting to connect..." << std::endl;
-    bool connected = sdk->Connect();
-    if (!connected) {
-        std::cout << "Could not connect to the sidecar. Exiting!" << std::endl;
-        return -1;
-    }
-    std::cout << "...handshake complete." << std::endl;
-
-    std::thread health (doHealth, sdk);
-    std::thread watch (watchUpdates, sdk);
-
-    std::cout << "Setting a label" << std::endl;
-    grpc::Status status = sdk->SetLabel("test-label", "test-value");
-    if (!status.ok()) {
-        std::cout << "Could not run SetLabel(): "+ status.error_message() + ". Exiting!" << std::endl;
-        return -1;
-    }
-
-    std::cout << "Setting an annotation" << std::endl;
-    status = sdk->SetAnnotation("test-annotation", "test value");
-    if (!status.ok()) {
-        std::cout << "Could not run SetAnnotation(): "+ status.error_message() + ". Exiting!" << std::endl;
+  std::cout << "C++ Game Server has started!" << std::endl;
+
+  std::cout << "Getting the instance of the SDK!" << std::endl;
+  agones::SDK *sdk = new agones::SDK();
+
+  std::cout << "Attempting to connect..." << std::endl;
+  bool connected = sdk->Connect();
+  if (!connected) {
+    std::cout << "Could not connect to the sidecar. Exiting!" << std::endl;
+    return -1;
+  }
+  std::cout << "...handshake complete." << std::endl;
+
+  std::thread health(doHealth, sdk);
+  std::thread watch(watchUpdates, sdk);
+
+  std::cout << "Setting a label" << std::endl;
+  grpc::Status status = sdk->SetLabel("test-label", "test-value");
+  if (!status.ok()) {
+    std::cout << "Could not run SetLabel(): " + status.error_message() +
+                     ". Exiting!"
+              << std::endl;
+    return -1;
+  }
+
+  std::cout << "Setting an annotation" << std::endl;
+  status = sdk->SetAnnotation("test-annotation", "test value");
+  if (!status.ok()) {
+    std::cout << "Could not run SetAnnotation(): " + status.error_message() +
+                     ". Exiting!"
+              << std::endl;
+    return -1;
+  }
+
+  std::cout << "Marking server as ready..." << std::endl;
+  status = sdk->Ready();
+  if (!status.ok()) {
+    std::cout << "Could not run Ready(): " + status.error_message() +
+                     ". Exiting!"
+              << std::endl;
+    return -1;
+  }
+
+  std::cout << "...marked Ready" << std::endl;
+
+  std::cout << "Getting GameServer details..." << std::endl;
+  agones::dev::sdk::GameServer gameserver;
+  status = sdk->GameServer(&gameserver);
+
+  if (!status.ok()) {
+    std::cout << "Could not run GameServer(): " + status.error_message() +
+                     ". Exiting!"
+              << std::endl;
+    return -1;
+  }
+
+  std::cout << "GameServer name: " << gameserver.object_meta().name()
+            << std::endl;
+
+  for (int i = 0; i < 10; i++) {
```

Comment:
Is this loop needed? The alternative would be something like:

```cc
// No loop
std::cout << "Sleeping for 1 minute... \n";
std::this_thread::sleep_for(std::chrono::minutes(1));
std::cout << "Shutting down...\n";
auto status = sdk.Shutdown();
...
```

Thoughts?

Reply:
I thought the same thing when I first saw it. There might still be a simpler way to re-write it, but what the loop does it to print out a message every 10 seconds (which is useful when you are debugging) and call shutdown after 60 seconds. The loop keeps running until 100 seconds, so you can see the difference between when shutdown is executed and when agones actually cleans up the container. I'm happy to rewrite it, but I think we should discuss in a different PR. Also note that other simple examples (rust at least) basically copied this code and if we change this one we should update the others as well.

Reply:
I still think we can make this quite a bit simpler, but it's not urgent.